cpp/src/arrow/dataset/file_base.h

```diff
@@ -114,6 +114,9 @@ class ARROW_DS_EXPORT FileSource {
   Result<std::shared_ptr<io::InputStream>> OpenCompressed(
       util::optional<Compression::type> compression = util::nullopt) const;
 
+  /// \brief equality comparison with another FileSource
+  bool Equals(const FileSource& other) const;
```

Comment:
We may want to inherit from EqualityComparable https://github.com/apache/arrow/blob/master/cpp/src/arrow/util/compare.h

Reply:
👍

Reply:
I added it, is it okay?

Reply:
I don't see the change for FileSource.

Reply:
Ah I missed that. I will add this too.